### PR TITLE
Add setting unwantedControlPanelsFields and use it in the function filterControlPanelsSchema

### PR DESCRIPTION
**Pull request by wesleybl**
`filterControlPanelsSchema` has been migrated to `Controlpanel.jsx` and uses the `unwantedControlPanelsFields` setting. So, if any portal wants to add fields in the Control Panel, it will be necessary to customize only the `unwantedControlPanelsFields` configuration, instead of customizing the entire `filterControlPanelsSchema` function.

This is an improvement for #3904

### CODE REVIEW

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:203]: Keep this. See next comment for reason.

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:204]: I'm not sure what version number this will get released into, so I put the next unreleased version number for now. It might need to be updated when it is actually released.

````suggestion
    ```{deprecated} Volto 16.20.8
    `filterControlPanelsSchema` was renamed to `unwantedControlPanelsFields` in Volto 16.20.8.
    ```
````

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:204]: This also might need a version number tweak at release.

````suggestion
    Control panels fields that are not used on the Volto.
    It is used internally to tweak the schemas provided by the `controlpanel` endpoint, to make them fit for Volto.

    ```{versionchanged} Volto 16.20.8
    `filterControlPanelsSchema` was renamed to `unwantedControlPanelsFields` in Volto 16.20.8.
    ```
````

File: news/4819.feature
Comment (stevepiercy) [inline news/4819.feature:1]:
```suggestion
Renames `filterControlPanelsSchema` setting to `unwantedControlPanelsFields`. @wesleybl
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Thanks for updating docs! 🤗

I reviewed docs only. A maintainer should review the source code.

**Review by davisagli — CHANGES_REQUESTED** [review]
Removing support for a custom `filterControlPanelsSchema` function is a breaking change that could only happen in a new major version of volto. I'd suggest keeping it, but change the default implementation to use the new `unwantedControlPanelFields` setting. Then it can be backported for volto 16.

**Comment by wesleybl** [comment]
> Removing support for a custom filterControlPanelsSchema function is a breaking change that could only happen in a new major version of volto.

@davisagli but isn't the master branch for Volto 17, which is a breaking release? Can't we keep `filterControlPanelsSchema` only on branch 16? If we don't remove it now, when will we remove it? How will we remember to remove it in the future?

**Comment by davisagli** [comment]
@wesleybl Yeah, we can make this change for 17. In general my bias is to keep backwards compatibility if it's easy. But that does inflate the number of options over time, and in this particular case I doubt the filterControlPanelsSchema setting is being used much. Any preference, @sneridagh?

**Comment by sneridagh** [comment]
@wesleybl I'm with @davisagli in this one... agreed that it cleans up the implementation, but that is hardly a reason for introducing a breaking if there is not delivering a clear direct advantage over the previous one.

**Comment by wesleybl** [comment]
@davisagli @sneridagh @stevepiercy I kept the `filterControlPanelsSchema` function and used `unwantedControlPanelsFields` in it.

Can you take a look please?

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:209]:
```suggestion
    A schema factory for a control panel.
    It is used internally, to tweak the schemas provided by the `@controlpanels` endpoint, making them fit for Volto.
    It uses the `unwantedControlPanelsFields` setting.
```

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:212]:
```suggestion
    Control panels fields that are not used in Volto.
    It is used internally by the `filterControlPanelsSchema` function.
```

File: packages/volto/news/4819.feature
Comment (stevepiercy) [inline packages/volto/news/4819.feature:1]: FYI, MyST uses single backticks for inline literals, whereas reStructuredText uses double backticks.

```suggestion
Add setting `unwantedControlPanelsFields` and use it in the function `filterControlPanelsSchema`. @wesleybl
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: docs/source/configuration/settings-reference.md
Comment (wesleybl) [reply docs/source/configuration/settings-reference.md]: Done.

File: packages/volto/news/4819.feature
Comment (wesleybl) [reply packages/volto/news/4819.feature]: > FYI, MyST uses single backticks for inline literals, whereas reStructuredText uses double backticks.

Thanks for the info! Done.

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:214]:
```suggestion
    Control panel fields that are not used in Volto.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
One minor grammar correction to the narrative docs made me notice that there might be a grammatical mistake in the naming of the schema factory. Why do use `filterControlPanelsSchema` and `unwantedControlPanelsFields` instead of `filterControlPanelSchema` and `unwantedControlPanelFields`? Obviously it is not worth refactoring the deprecated `filterControlPanelsSchema`, but it might be worth refactoring the new factory to `unwantedControlPanelFields`. I understand that the endpoint is `@controlpanels`. Thoughts?

File: docs/source/configuration/settings-reference.md
Comment (wesleybl) [reply docs/source/configuration/settings-reference.md:214]: The fields can be from several control panels. For example: navigation, site...

That's why it's in the plural. @davisagli  what is your opinion here?

**Comment by wesleybl** [comment]
> I understand that the endpoint is @controlpanels. Thoughts?

Yes: https://github.com/plone/plone.restapi/blob/11275deb4bc9488e0c10929bd7c76a1cfd6edc96/src/plone/restapi/services/controlpanels/configure.zcml#L32

File: docs/source/configuration/settings-reference.md
Comment (stevepiercy) [reply docs/source/configuration/settings-reference.md:214]: Ah, in that case, it should be something else, using the plural possessive form. I edited my suggestion.

**Comment by stevepiercy** [comment]
@wesleybl let's leave it as is without a refactor. Plural makes sense. I made a minor change to my suggestion, and if you accept, then I can approve and move this forward. Thank you!

**Review by stevepiercy — APPROVED** [review]
Thank you!

**Review by davisagli — APPROVED** [review]
This looks good to me now.

**Comment by wesleybl** [comment]
@davisagli @ichim-david I updated the code and resolved the conflicts. Can we merge it?

**Review by davisagli — APPROVED** [review]
Looks good to me.

**Pull request merged**
